packages/schematics/src/collection/library/index.ts

```diff
@@ -225,16 +225,23 @@ function updateNgPackage(options: NormalizedSchema): Rule {
   ]);
 }
 
-function updateProject(options: NormalizedSchema): Rule {
+function updateProject(options: NormalizedSchema, publishable = false): Rule {
   return (host: Tree) => {
-    // Bug in @angular-devkit/core. Cannot delete these files here.
-    // host.delete(`${options.projectRoot}/src/lib/${options.name}.service.ts`);
-    // host.delete(`${options.projectRoot}/src/lib/${options.name}.service.spec.ts`);
-    // host.delete(`${options.projectRoot}/src/lib/${options.name}.component.ts`);
-    // host.delete(`${options.projectRoot}/src/lib/${options.name}.component.spec.ts`);
+    const libRoot = `${options.projectRoot}/src/lib/`;
+
+    host.delete(path.join(libRoot, `${options.name}.service.ts`));
+    host.delete(path.join(libRoot, `${options.name}.service.spec.ts`));
+    host.delete(path.join(libRoot, `${options.name}.component.ts`));
+    host.delete(path.join(libRoot, `${options.name}.component.spec.ts`));
+
+    if (!publishable) {
```

Comment:
We can read publishable from options:
```ts
if (!options.publishable) {
```

Reply:
NormalizedSchema extends Schema, which has publishable